src/tests/conformance/messages.rs

```diff
@@ -178,15 +177,102 @@ async fn ignores_unsolicited_responses() {
             .await
             .unwrap();
 
-        match filter.read_from_stream(&mut stream).await.unwrap() {
-            Message::Pong(returned_nonce) => assert_eq!(nonce, returned_nonce),
-            msg => panic!("Expected pong: {:?}", msg),
-        }
+        let pong = filter.read_from_stream(&mut stream).await.unwrap();
+        assert_eq!(pong.kind(), MessageKind::Pong);
     }
 
     node.stop().await;
 }
 
+#[tokio::test]
+async fn basic_query_response() {
+    // ZG-CONFORMANCE-010
+    //
+    // The node responds with the correct messages. Message correctness is naively verified through successful encoding/decoding.
```

Comment:
This test is mostly covered by more specific cases but I think it has value in testing explicit expectations around how a node should respond to queries (at least while the spec isn't clearly defined)?

Reply:
It is certainly redundant in terms of coverage. But I like having it as it as it highlights discrepancies between responses.

I think we may want to refactor our test spec and contents once we have the spec more nailed down.

Reply:
Agreed 👍